Geometry/HcalCommonData/src/HcalDDDRecConstants.cc

```diff
@@ -448,29 +468,28 @@ HcalDDDRecConstants::HcalCellTypes(HcalSubdetector subdet) const {
     int isub   = (subdet == HcalBarrel) ? 0 : 1;
     std::vector<HcalDDDRecConstants::HcalEtaBin> etabins = getEtaBins(isub);
     std::vector<int> missPhi;
-    for (unsigned int bin=0; bin<etabins.size(); ++bin) {
+    for (auto etabin : etabins) {
```

Comment:
In that case, it should still be `auto&`, unless the changes are intentionally made to a copy of the variable. Also, I don't actually see any places in this function where `etabin` is changed...